### PR TITLE
Changed requirements in the recipes will overwrite automatic requirement updates

### PR DESCRIPTION
**Pull request by critias**
- Some cleanup of the automatic requirement update 
- Changed requirements in the recipes will now overwrite automatic requirement updates.

### CODE REVIEW

**Comment by michelwi** [comment]
> * Changed requirements in the recipes will now overwrite automatic requirement updates.

Does this mean once I specify a custom rqmt in my recipe, the actual rqmt isn't doubled any more in case of oom/oot?

Or does it mean if e.g. (i first define 1G of mem rqmt, my job crashes with oom, then i update the recipe to 8G) then the job is resubmitted with 8G rather than 2x 1G?

File: sisyphus/global_settings.py [hunk 1]
```diff
@@ -58,46 +59,33 @@ def worker_wrapper(job, task_name, call):
     return call
 
 
-def update_engine_rqmt(initial_rqmt, last_usage):
-    """ Update requirements after a job got interrupted.
+def update_engine_rqmt(last_rqmt: Dict, last_usage: Dict):
+    """ Update requirements after a job got interrupted, double limits if needed
 
-    :param dict[str] initial_rqmt: requirements that are requested first
+    :param dict[str] last_rqmt: requirements that are requested first
```
Comment (patrick-wilken) [inline sisyphus/global_settings.py:65]: Doc string should be something like: requirements requested for the previous run (possibly too little time or memory)

**Comment by patrick-wilken** [comment]
> Or does it mean if e.g. (i first define 1G of mem rqmt, my job crashes with oom, then i update the recipe to 8G) then the job is resubmitted with 8G rather than 2x 1G?

Yes, this. Before, if the submit_log file still existed the last entry would be read and would overwrite the requirements set in the recipes, even if you intentionally updated them yourself to avoid OOM. Now we check whether these user defined requirements changed and if yes always use them. Only in the other case this automatic doubling of requirements is enabled (possibly multiple times if it keeps failing).

Logic looks good to me. I guess we can try it for a few days and then merge?

**Review by patrick-wilken — APPROVED** [review]

**Comment by patrick-wilken** [comment]
I realized only now that updating the requirements is also needed when excluding faulty cluster nodes. That finally worked for me right away as expected. And had no other problems using this branch.

**Pull request merged**